src/plugins/ui_core_views/evaluation_conditional_format.ts

```diff
@@ -366,13 +367,24 @@ export class EvaluationConditionalFormatPlugin extends CoreViewPlugin {
     const { sheetId } = target;
     const evaluator = criterionEvaluatorRegistry.get(rule.operator);
 
+    let hasMultipleElementMatrix = false;
     const evaluatedCriterionValues = rule.values.map((value) => {
       if (!value.startsWith("=")) {
         return parseLiteral(value, DEFAULT_LOCALE);
       }
-      return this.getters.evaluateFormula(sheetId, value) ?? "";
+      const result = this.getters.evaluateFormula(sheetId, value);
```

Comment:
It looks like something like this would also work and is shorter.
It also allows to remove the type casting

```diff
@@ -372,13 +372,13 @@ export class EvaluationConditionalFormatPlugin extends CoreViewPlugin {
       }
       return this.getters.evaluateFormula(sheetId, value) ?? "";
     });
-    if (evaluatedCriterionValues.some(isMatrix)) {
+    if (evaluatedCriterionValues.some(isMultipleElementMatrix)) {
       return false;
     }
 
     const evaluatedCriterion = {
       type: rule.operator,
-      values: evaluatedCriterionValues as CellValue[],
+      values: evaluatedCriterionValues.map(toScalar),
     };
     return evaluator.isValueValid(cell.value ?? "", evaluatedCriterion, this.getters, sheetId);
   }
```